src/library_pthread.js

```diff
@@ -747,7 +755,6 @@ var LibraryPThread = {
   },
 
   // Public pthread_self() function which returns a unique ID for the thread.
-  pthread_self__asm: true,
```

Comment:
My original intent I believe was to change behavior as little as possible. On incoming as of today, this is not included in the asm.js portion.
Adding `__sig: 'i'`seems to work, but I'm not fully sure why. `pthread_self` depends on the library-provided value `__pthread_ptr`, which isn't found in the asm.js...

Ok after investigating that, this can't currently be asm because it depends on `__pthread_ptr`, which we as far as I know we can't currently express should be part of the module. In principle we can? But we currently don't.

Although `_pthread_ptr` is also used in `_register_pthread_ptr`, which is called from various places as well. I'll remove the untrue `__asm` from `_register_pthread_ptr` as well.

So, I'm going to leave `pthread_self` and `_register_pthread_ptr` where they currently are before this PR (outside the asm module, called via FFI), and as a future project we can figure out how to migrate them in.